spotlight-client/src/charts/ResponsiveTooltipController.tsx

```diff
@@ -0,0 +1,176 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2021 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import useBreakpoint from "@w11r/use-breakpoint";
+import isEmpty from "lodash.isempty";
+import React, { useEffect, useState } from "react";
+import { AnnotationType } from "semiotic/lib/types/annotationTypes";
+import { XYFrameProps } from "semiotic/lib/types/xyTypes";
+import { OrdinalFrameProps } from "semiotic/lib/types/ordinalTypes";
+import {
+  TooltipContentProps,
+  useInfoPanelDispatch,
+  useInfoPanelState,
+} from "../InfoPanel";
+import Tooltip, { TooltipProps } from "../Tooltip";
+import { isItemToHighlight, ItemToHighlight } from "./types";
+
+function chartDataToTooltipProps({ label, value, pct }: TooltipContentProps) {
+  return {
+    title: label as string,
+    records: [
+      {
+        value: value as number,
+        pct: pct as number,
+      },
+    ],
+  };
+}
+
+type SemioticChildProps = Partial<XYFrameProps> | Partial<OrdinalFrameProps>;
+
+export type ResponsiveTooltipControllerProps = {
+  customHoverBehavior?: (record: TooltipContentProps) => void;
+  getTooltipProps?: (record: TooltipContentProps) => TooltipProps;
+  hoverAnnotation?:
+    | XYFrameProps["hoverAnnotation"]
+    | OrdinalFrameProps["hoverAnnotation"];
+  pieceHoverAnnotation?: boolean;
+  render?: (props: SemioticChildProps) => React.ReactElement;
+  setHighlighted?: (item?: ItemToHighlight) => void;
+};
+
+/**
+ * a wrapper around visualization components that passes props
+ * as needed for native hover annotations in Semiotic but also
+ * uses same configuration to power our custom InfoPanel component
+ * (which is for touch screens, which Semiotic does not natively support
```

Comment:
This stuff all came along for the ride because it was a dependency of the chart (which, in retrospect, I kind of regretted because it ballooned the size of this PR so much) ... nothing really works on mobile yet because the page layout is not adaptive, but the good news is that once I figure that out, the existing chart functionality should pretty much Just Work. so i guess uh .... I am asking you to trust me on this for now because I don't want to expand the scope of this PR any further? sorry that kinda sucks

Reply:
see #309 wherein all of this stuff will actually be fixed